sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -149,10 +150,20 @@ case class AggregateExpression(
       case PartialMerge => "merge_"
       case Final | Complete => ""
     }
-    prefix + aggregateFunction.toAggString(isDistinct)
+    val aggFuncStr = prefix + aggregateFunction.toAggString(isDistinct)
+    filter match {
+      case Some(predicate) => s"$aggFuncStr filter (where $predicate)"
```

Comment:
If you don't mind, `filter (where` -> `FILTER (WHERE`?

Currently, in the generated result, `filter (where NOT` and `FILTER (WHERE (NOT` are used together. Maybe, the same output might be better?
> sum(salary#x) filter (where NOT exists#x [dept_id#x]) AS sum(salary) FILTER (WHERE (NOT exists(dept_id)))#x]

Reply:
yea, ok.